test/integration/thread_arena.c

```diff
@@ -37,24 +37,44 @@ thd_start(void *arg) {
 	return NULL;
 }
 
+void
+mallctl_failure(int err)
+{
+	char buf[BUFERROR_BUF];
+
+	buferror(err, buf, sizeof(buf));
+	test_fail("Error in mallctl(): %s", buf);
+}
+
 TEST_BEGIN(test_thread_arena) {
 	void *p;
 	unsigned arena_ind;
-	size_t size;
+	size_t size, size_opt_percpu_arena;
 	int err;
 	thd_t thds[NTHREADS];
 	unsigned i;
+	const char *opt_percpu_arena;
 
 	p = malloc(1);
 	assert_ptr_not_null(p, "Error in malloc()");
 
+	size_opt_percpu_arena = sizeof(opt_percpu_arena);
+	if ((err = mallctl("opt.percpu_arena", &opt_percpu_arena,
+	    &size_opt_percpu_arena, NULL, 0))) {
+		mallctl_failure(err);
+		goto label_done;
+	}
+	if (!strcmp(opt_percpu_arena, "disabled")) {
+		/* Skip if perCPU arena is enabled. */
+		test_skip("\"opt.percpu_arena\" enabled");
```

Comment:
```test_skip_if()``` should be fine here (move ```malloc()``` call further down).  FWIW, the tests don't generally try hard to clean up (nor even keep from crashing) if a failure occurs, so feel free to write less robust error recovery.